packages/flutter/lib/src/rendering/binding.dart

```diff
@@ -22,28 +21,34 @@ export 'package:flutter/gestures.dart' show HitTestResult;
 // Examples can assume:
 // late BuildContext context;
 
-/// The glue between the render tree and the Flutter engine.
+/// The glue between the render trees and the Flutter engine.
+///
+/// The [RendererBinding] manages multiple render trees. Each render tree is
+/// rooted in a [RenderView] that must be added to the binding via
+/// [addRenderView] to be considered during frame production, hit testing, etc.
+/// Furthermore, the render tree must be managed by a [PipelineOwner] that is
+/// part of the pipeline owner tree rooted at [rootPipelineOwner].
+///
+/// Adding [PipelineOwner]s and [RenderView]s to this binding in the way
+/// described above is left as a responsibility for a higher level abstraction.
+/// The widgets library, for example, introduces the [View] widget, which
+/// registers its [RenderView] and [PipelineOwner] with this binding.
```

Comment:
```suggestion
/// The [RendererBinding] manages a array of render trees. Each render tree is
/// rooted in a [RenderView]. The [RendererBinding] handles the render views'
/// frame production, hit testing, etc.
///
/// A render tree must also be managed by a [PipelineOwner]. The
/// [PipelineOwner]s constitute a pipeline owner tree that is rooted at
/// [rootPipelineOwner], which is also managed by the [RendererBinding].
///
/// After adding a render view to the binding via [addRenderView], the caller 
/// must also add a [PipelineOwner] to the pipeline owner tree.  Ensuring this
/// consistency is left as a responsibility for a higher level abstraction, such
/// as the [View] widget.
```

Reply:
I am not convinced by this suggested change. 😄 What do you dislike about the original doc?

> The [RendererBinding] manages a array of render trees.

I don't want to make any promises about implementation details (i.e. the render trees aren't really managed in an array and it's also not relevant information).

> After adding a render view to the binding via [addRenderView]

The order this has to happen in isn't important. Either order is fine.

> Ensuring this consistency is left as a responsibility for a higher level abstraction, such as the [View] widget.

To me, this has the wrong focus. The main take away here should be that somebody has to add render views and pipeline owners to this binding if they want the binding to manage them. The binding isn't creating these entities for you. It isn't really about consistency (consistency about what, BTW?).

Reply:
I mostly wanted to add "The [RendererBinding] manages a array of render trees", while the rest is basically wording adjustment that seems easier to understand to me.

> I don't want to make any promises about implementation details (i.e. the render trees aren't really managed in an array and it's also not relevant information).

I wanted to emphasize that the render trees are not related to each other, unlike the pipeline owners, which are structured in a tree. The "array" here is not about the implementation, but about the abstract model. It really interests me that a view corresponds to two things that are structured differently, and I think emphasizing it helps understanding the model.

> It isn't really about consistency (consistency about what, BTW?).

I might have used the wrong words... I wanted to emphasize the contract that adding a renderview must correspond to adding a pipeline owner as well.

Reply:
What do you think of this? 
```dart
/// The [RendererBinding] manages render trees. Each render tree is rooted in a
/// [RenderView]. The [RendererBinding] handles the render views' frame
/// production, hit testing, etc.
/// 
/// A render tree must also be managed by a [PipelineOwner]. The
/// [PipelineOwner]s constitute a pipeline owner tree that is rooted at
/// [rootPipelineOwner], which is also managed by the [RendererBinding].
/// 
/// To add a render view to the binding, the caller must call [addRenderView] 
/// and also add a [PipelineOwner] to the pipeline owner tree (in any order).
/// Ensuring this contract is left as a responsibility for a higher level
/// abstraction, such as the [View] widget.
```

Reply:
> What do you dislike about the original doc?

I'm proposing to split it into three paragraphs:
1. The binding manages render trees. (Avoid talking about the pipeline owner, and the lifecycle of render trees)
2. Introduce the pipeline owner (Avoid the lifecycle management)
3. Lifecycle management.

I personally find it clearer to understand it this way.